block_course_contents.php

```diff
@@ -164,15 +164,41 @@ public function get_content() {
                 $title = $format->get_section_name($section);
             }
 
+            if ( ($i == 0) && (!empty($globalconfig->display_course_link)) ) {
+                $sectionclass = 'section-item';
+
+                if ((!isset($selected)) && (empty($selected)) ) {
+                    $sectionclass .= ' current ' . $selected;
+                }
+                $text .= html_writer::start_tag('li', array('class' => $sectionclass));
+
+                $text .= html_writer::span('>', 'section-number');
+                if (!empty($globalconfig->display_course_link_text)) {
+                    $anchortext = $globalconfig->display_course_link_text;
+                } else {
+                    $anchortext = $course->shortname;
+                }
+
+                if ((!isset($selected)) && (empty($selected)) ) {
+                    $text .= '&nbsp; ' . $anchortext;
+                } else {
+                    $text .= '&nbsp; ' . html_writer::link(course_get_url($course), $anchortext);
+                }
+
+                $text .= html_writer::end_tag('li');
+            }
+
             $odd = $r % 2;
-            if ($format->is_section_current($section)) {
-                $text .= html_writer::start_tag('li', array('class' => 'section-item current r'.$odd));
+            if (isset($selected) && $i == $selected) {
+                $text .= html_writer::start_tag('li', array('class' => 'section-item current r.$odd'));
             } else {
                 $text .= html_writer::start_tag('li', array('class' => 'section-item r'.$odd));
             }
 
-            if ($i == 0) {
-                // Never enumerate the section number 0.
+            $enumeratesection0 = (!empty($globalconfig->enumerate_section_0) ? true : false);
```

Comment:
Again, this should be instance-level setting. Also, I usually prefer explicit if/then block rather than this.